engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -820,26 +901,38 @@ class Compiler(
     */
   private def runMethodBodyPasses(
     ir: IRModule,
-    moduleContext: ModuleContext
+    moduleContext: ModuleContext,
+    irDumper: Option[IRDumper]
   ): IRModule = {
     context.log(
       Level.FINEST,
       "Passing module {0} with method body passes",
       moduleContext.module.getName
     )
-    passManager.runPassesOnModule(ir, moduleContext, passes.functionBodyPasses)
+    passManager.runPassesOnModule(
+      ir,
+      moduleContext,
+      passes.functionBodyPasses,
+      irDumper
+    )
   }
 
   private def runGlobalTypingPasses(
     ir: IRModule,
-    moduleContext: ModuleContext
+    moduleContext: ModuleContext,
```

Comment:
`ModuleContext` references `CompilerConfig` - the `IRDumper` to use should be provided by `CompilerConfig` I think.